hazelcast-client/src/main/java/com/hazelcast/client/spi/impl/ClientInvocationFuture.java

```diff
@@ -164,7 +166,20 @@ private V resolveResponse() throws ExecutionException, TimeoutException, Interru
         if (response == null) {
             throw new TimeoutException();
         }
-        return (V) response;
+        return (V) verifyResponse(response);
+    }
+
+    private Object verifyResponse(Object value) {
+        if (value instanceof Data) {
+            return value;
+        }
+        if (operationResultVerifier != null) {
+            value = operationResultVerifier.verify((V) value);
```

Comment:
I already read the `javax.cache.Cache Javadoc`. I mean, why you just directly don't throw `ClassCastException` while verifying ? Why you return exception as return value then check if it `Throwable` then throw ?